Fix output of filtering module to avoid overwriting (#37)
qbic-pipelines/vcftomat: docs/output.md

- [MultiQC](#multiqc) - Aggregate report describing results and QC from the whole pipeline
- [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution

> The `${meta.id}` is the sample id given in the samplesheet.
> 
> ### Tabix
> 
> Tabix generated index files with `.tbi` extension for all `(g).vcf` files that are given to the pipeline without index.
> 

Author: ```suggestion

### Tabix

Tabix generated index files with `.tbi` extension for all `(g).vcf` files that are given to the pipeline without index.

<details markdown="1">
<summary>Output files</summary>

- `tabix`
  - `{filename}.vcf.gz.tbi`: tabix index of the vcf file.

</details>
```

### Filter VCFs

<details markdown="1">
<summary>Output files</summary>

- `bcftools/view/{meta.label}/`
  - `{filename}.filter.vcf.gz`: vcf file with filtered variants.
  - `{filename}.filter.vcf.gz.tbi`: tabix index of the vcf file.

</details>

VEP annotated VCF files can be filtered for certain flags present after VEP annotation. Notably, this enables filtering for variants with certain impact levels or consequences. Filtering will produces VCF files holding just the variants matching the specific patterns.

### Concatenate VCFs

<details markdown="1">
<summary>Output files</summary>

- `bcftools/concat/`
  - `{sample}.{label}.concat.vcf.gz`: vcf file containing all variants from files with same {sample} and {label}.
  - `{sample}.{label}.concat.vcf.gz.tbi`: tabix index of the vcf file.

</details>

Some variant calling pipelines will return multiple (g)VCF files for one patient. The `concatenate` function of `bcftools` is used to add these VCFs to one VCF.

### Rename Samples

<details markdown="1">
<summary>Output files</summary>

> - `bcftools/reheader/${meta.id}/`

Author: ```suggestion
- `bcftools/reheader/{sample}/`
```

- [MultiQC](#multiqc) - Aggregate report describing results and QC from the whole pipeline
- [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution

### Tabix

Tabix generated index files with `.tbi` extension for all `(g).vcf` files that are given to the pipeline without index.

<details markdown="1">
<summary>Output files</summary>

- `tabix`
  - `{filename}.vcf.gz.tbi`: tabix index of the vcf file.

> </details>

Author: ```suggestion
</details>

```

- [MultiQC](#multiqc) - Aggregate report describing results and QC from the whole pipeline
- [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution

> 

Author: ```suggestion
```

Tabix generated index files with `.tbi` extension for all `(g).vcf` files that are given to the pipeline without index.

<details markdown="1">
<summary>Output files</summary>

- `tabix`
  - `{filename}.vcf.gz.tbi`: tabix index of the vcf file.

</details>

> 

Author: ```suggestion

<details markdown="1">
<summary>Output files</summary>

- `tabix`
  - `{filename}.vcf.gz.tbi`: tabix index of the vcf file.

</details>

Tabix generated index files with `.tbi` extension for all `(g).vcf` files that are given to the pipeline without index.

```

The GATK GenotypeGVCFs module translates genotype (g) vcf files into classic vcf files. The key difference between a regular VCF and a GVCF is that the GVCF has records for all sites, whether there is a variant call there or not.

> The output directory is called `gvcftovcf/{meta.label}`.

Reviewer: Usually at least important output files are explained like for the MultiQC files, but at least the directory is mentioned now, so that progress ;)

Author: I was a bit lazy to write all these infos, but maybe I'll take the opportunity of this PR to finally do it :D

Reviewer: I once run a pipeline and it was not clear which file in the vast output directory held the final result. The authors thankfully did respond to an email, but did not clarify appropriately. Therefore the pipeline was useless for me.
